### DISCUSSION TITLE
How can I see where my ports are in the gds displayed in KLayout?

### DISCUSSION THREAD

**Discussion by QuentinWach** [gdsfactory/gdsfactory#3379] [Q&A] [answered]
Nazca provides the feature that ports are marked with circles and labels directly in the gds file in a way that scales with window so that they can be easily found on a first glance when looking at a gds in Nazca. Is there such a feature in GDSFactory or how could I do this?

**Comment by sebastian-goeldi** [comment, marked as answer]
Please install the [gdsfactory klayout extension](http://sami.klayout.org/preview?url=git%2Bhttps%3A%2F%2Fgithub.com%2Fgdsfactory%2Fgdsfactory.git%2Fbranches%2Fmain%2Fgdsfactory%2Fgeneric_tech%2Fklayout) to make it possible. You can find this in Tools > Manage Packages (or Package Manager, forgot the exact name sorry). You will get a button `Show Ports` in klayout. On click it will togle ports for the cureently shown cell. In the dropdown there is an option with toggles for all or individual cells.

gdsfactory (or to be exact the underlying kfactory) writes ports and other additional informatin as klayout MetaInfo objects into the gds/oasis.

A lot of fab tools are not compatible with this, so you can drop it with `write_gds(with_metadata=False)`

**Reply by QuentinWach** [reply]
Thank you, Sebastian. Exactly what I was searching for :)

**Comment by joamatab** [comment]
+1 on Sebastian's suggestion

you can also do

```python
c = gf.c.straight()
c.draw_ports()
c.show()
```
